### ISSUE TITLE
Numpy dtype float deprecated

### ISSUE THREAD

**Issue by Tdarnell** [GeosoftInc/gxpy#128]
Opening a gdb file using geosoft-2022.2.0 results in an attribute error, due to the deprecation of float as a numpy type in versions >1.20

Python 3.10.11 | packaged by Anaconda, Inc. | (main, Apr 20 2023, 18:56:50) [MSC v.1916 64 bit (AMD64)]
certifi-2023.5.7
charset-normalizer-3.1.0
geosoft-2022.2.0
idna-3.4
numpy-1.24.3
pandas-2.0.1
python-dateutil-2.8.2
pytz-2023.3
requests-2.30.0
six-1.16.0
tzdata-2023.3
urllib3-2.0.2

```
File [.conda_venv\lib\site-packages\geosoft\gxpy\gdb.py:1782](file:///.conda_venv/lib/site-packages/geosoft/gxpy/gdb.py:1782), in Geosoft_gdb.read_line(self, line, channels, dtype, fid, dummy)
   1780 w = self.channel_width(cs)
   1781 if w == 1:
-> 1782     vv = self.read_channel_vv(ls, cs, dtype=npd.dtype)
   1783     if vv.length > 0:
...

AttributeError: module 'numpy' has no attribute 'float'.
`np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

**Comment by serban-seeq** [comment]
This will be fixed in next release. Until then please force numpy to 1.19

`pip install numpy==1.19.0`

GXDEV-34

**Comment by ianneilmacleod** [comment]
This has indeed been fixed in the 2023.11.21, though only for the Geosoft embedded Python.  If you are using
your own Python environment you will still need to either depricate to an older version of Numpy, or modify the 
utilities.py file yourself.

In my case, I have my own Python 3.8 envoronment into which I installed the last Geosoft (pip install geosoft).  You can edit the "...\envs\Python38\Lib\site-packages\geosoft\gxpy\utility.py" to comment out the lines that include 'np.float' and 'np.int', which are both no longer used and not necessary in the mapping statements in which they appear.  If this does not make sense to you then you probably should not be messing with it.  For those that are familiar with github, I forked Geosoft's gxpy and made the modifications there. You will find these in my forked gxpy here - https://github.com/ianneilmacleod/gxpy/blob/master/geosoft/gxpy/utility.py.

Note - if you are reading this, I am long retired from Geosoft and only recently came back to Python to work with some DEM files so I can make relief carvings on my CNC machine. I don't do geoscience things any longer, but it is fun to mess with python again.  Lots of good stuff in the Geosoft API (IMHO).

**Comment by TheZappie** [comment]
Is this issue actually solved with release 2024.1.0 ? 

From a fresh install of python 3.12 _pip install numpy geosoft_ gives versions:
numpy                     2.0.0
geosoft                   2024.1.0

The file "...Lib\site-packages\geosoft\gxpy\utility.py"
still contains "np.float_" which will crash with numpy 2.x

**Comment by serban-seeq** [comment]
@TheZappie firstly, as a disclaimer, the 2024.1 hasn't been officially release, and wont be for about a month. 

That being said, the fix was to change `float` => `float_`. `float` does indeed crash numpy 2.x, but `float_` should work and is passing in our tests.

Can you provide a script and the error message with the crash you are seeing in 2024.1?

**Comment by TheZappie** [comment]
Error stack:

  File "D:\2-NonProject\MagVision\venv\Lib\site-packages\geosoft\gxpy\gdb.py", line 2143, in write_line
    self.write_channel(line, cs, data[:, np_index: np_index + w], fid=fid)
  File "D:\2-NonProject\MagVision\venv\Lib\site-packages\geosoft\gxpy\gdb.py", line 2024, in write_channel
    cs = self.new_channel(channel, data.dtype, array=_va_width(data))
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\2-NonProject\MagVision\venv\Lib\site-packages\geosoft\gxpy\gdb.py", line 1186, in new_channel
    gxu.gx_dtype(dtype),
    ^^^^^^^^^^^^^^^^^^^
  File "D:\2-NonProject\MagVision\venv\Lib\site-packages\geosoft\gxpy\utility.py", line 552, in gx_dtype
    str(np.dtype(np.float_)): gxapi.GS_DOUBLE,
                 ^^^^^^^^^
  File "D:\2-NonProject\MagVision\venv\Lib\site-packages\numpy\__init__.py", line 411, in __getattr__
    raise AttributeError(
AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?


if I do 
```
import numpy
np.float_
```
I get this error as well with numpy 2.0, maybe your tests were run with  a older version of numpy?

**Comment by serban-seeq** [comment]
Oh I see what's going on. This fix was to support numpy `1.20`+.
You are using numpy `2.0` which was released only 3 days ago so we didn't know about it at the time the release was made. Please downgrade to 1.26. I will review the logic for numpy dependency.

**Comment by TheZappie** [comment]
Thanks. 
Another option would be to make it actually numpy 2.x compatible :)
